src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -388,6 +389,25 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
     }
   }
 
+  /**
+   * Returns the target-specific execution platform constraints, based on the rule definition and
+   * any constraints added by the target.
+   */
+  private static ImmutableSet<Label> getExecutionPlatformConstraints(Rule rule) {
+    NonconfigurableAttributeMapper mapper = NonconfigurableAttributeMapper.of(rule);
+    ImmutableSet.Builder<Label> execConstraintLabels = new ImmutableSet.Builder<>();
+
+    execConstraintLabels.addAll(rule.getRuleClassObject().getExecutionPlatformConstraints());
+
+    if (rule.getRuleClassObject().executionPlatformConstraintsAllowed() == ExecutionPlatformConstraintsAllowed.PER_TARGET
```

Comment:
I haven't counted it by hand but this line looks like it might exceed 100 chars.

Reply:
Done.